docs/modules/record-encryption/con-record-encryption-overview.adoc

```diff
@@ -0,0 +1,95 @@
+// file included in the following:
+//
+// assembly-record-encryption-filter.adoc
+
+[id='con-topic-encryption-overview-{context}']
+= Encryption process
+
+[role="_abstract"]
+The Record Encryption filter uses envelope encryption to encrypt records with symmetric encryption keys. 
+The filter encrypts records from produce requests and decrypts records from fetch responses.
+
+Envelope encryption::
+Envelope encryption is an industry-standard technique suited for encrypting large volumes of data in an efficient manner.
+Data is encrypted with a Data Encryption Key (DEK).
+The DEK is encrypted using a Key Encryption Key (KEK).
+The KEK is stored securely in a Key Management System (KMS).
+Symmetric encryption keys::
+AES(GCM) 256 bit encryption symmetric encryption keys are used to encrypt and decrypt record data. 
+
+The process is as follows:
+
+. The filter intercepts produce requests from producing applications and encrypts the records.
+. The produce request is forwarded to the broker. 
+. The filter intercepts fetch responses from consuming applications and decrypts the records.
+. The fetch response is forwarded to the consuming application.
+
+The filter encrypts the record value only.
+Record keys, headers, and timestamps are not encrypted.
+
+The entire process is transparent from the point of view of Kafka clients and Kafka brokers. 
+Neither are aware that the records are being encrypted, nor do they have any access to the encryption keys or have any influence on the ciphering process to secure the records.
+
+== How the filter encrypts records
+The filter encrypts records from produce requests as follows:
+
+. Filter selects a KEK to apply.
+. Requests the KMS to generate a DEK for the KEK.
+. Uses an encrypted DEK (DEK encrypted with the KEK) to encrypt the record.
+. Replaces the original record with a cipher record (encrypted record, encrypted DEK, and metadata).
```

Comment:
Is it a `cipher record` or `enciphered record` ? I'm not sure what the correct/ commonly accepted term is.

Reply:
In the design we use the term "ciphertext record" to describe the record content that gets forwarded to the broker.   I think that's the term we should favour in the user facing documentation.

https://github.com/kroxylicious/kroxylicious/blob/4535cc01e7de2face81ea9d65827988f6d1c3a8a/kroxylicious-filters/kroxylicious-record-encryption/doc/design.adoc?plain=1#L231

Reply:
I'm all for consistency so `ciphertext record` would be the path of least resistance but still sounds a bit odd to me. 

The more I think about it the harder time I have coming up with a phrase that works for me, so lets just be consistent unless @PaulRMellor has better idea.